### ISSUE TITLE
Region Folding Via Indentation Only Works With Extra Newlines 

### ISSUE THREAD

**Issue by brycemorrow4564** [Lencerf/vscode-beancount#62]
Hello! First of all just want to say thanks for this plugin, it's been extremely helpful. 

I'm currently building out a beancount file, and I'm interested in organizing my sections via indentation using org mode style headers. I've noticed some strange behavior in vs code related to the interplay between syntax highlighting and indentation. I've included two images here which showcase the issue.

This first image shows the issue with syntax highlighting when there is there is a newline + a tab separating the two option directives: 
![Screen Shot 2021-10-01 at 11 44 26 AM](https://user-images.githubusercontent.com/9085296/135649301-80e4cadf-803c-4a32-812b-e04868889baa.png)

The second image shows that syntax highlighting works correctly when you add in one (or more) newlines between the option directives: 
![Screen Shot 2021-10-01 at 11 44 36 AM](https://user-images.githubusercontent.com/9085296/135649317-9a62c85c-cff2-4ca7-8c37-247820d71ccf.png)

Has anyone else experienced the same issue? Or could this possibly be due to some local misconfiguration on my part?

**Comment by Lencerf** [comment]
> syntax highlighting works correctly when you add in one (or more) newlines between the option directives:

Yes. According to  https://github.com/Lencerf/vscode-beancount/blob/533719ee72f97ccf10b448d69d12b049e52837f5/syntaxes/beancount.tmLanguage#L118

currently, an `option` directive must be followed by an empty newline, or a line starting with non-whitespace characters.

It is not hard to add syntax highlighting support to the code style in your first image, but beancount itself requires `option` directive lines start with no whitespaces. As I tested, `bean-report` complains `syntax error, unexpected INDENT` when there are spaces in front of `option`. Did you see similar errors?

**Comment by MatthewVasseur** [comment]
> As I tested, `bean-report` complains `syntax error, unexpected INDENT` when there are spaces in front of `option`. Did you see similar errors?

_(not the original poster but...)_ Agreed I see the same syntax error of `syntax error, unexpected INDENT`

That being said, as far as I can tell, this syntax error is the only thing that is broken from beancount. It's still able to parse the file and generate reports.

I'm curious whether those people controlling folding via indents (e.g. https://github.com/Lencerf/vscode-beancount/pull/15) have ignored these `bean-check` errors?
